README.md

```diff
@@ -0,0 +1,158 @@
+instrumentation rules for the Java `HttpURLConnection`.
+
+NOTE: Currently the agent instrumentation rules for installing the servlet filter are provided as a
+separate maven dependency. Eventually these rules will be bundled with the servlet integration project.
+
+### Add an OpenTracing Compliant Tracer
+
+This example has currently been configured to use the Hawkular APM OpenTracing compliant Tracer:
+
+```xml
+    <!-- OpenTracing compliant Tracer dependencies -->
+    <dependency>
+      <groupId>org.hawkular.apm</groupId>
+      <artifactId>hawkular-apm-client-opentracing</artifactId>
+      <version>...</version>
+    </dependency>
+    <dependency>
+      <groupId>org.hawkular.apm</groupId>
+      <artifactId>hawkular-apm-trace-publisher-rest-client</artifactId>
+      <version>...</version>
+    </dependency>
+```
+
+However any suitable `Tracer` implementation can be used, as long as it can be located using the
+[Global Tracer](https://github.com/opentracing-contrib/java-globaltracer) utility.
+
+Use of Hawkular APM's Tracer also requires a custom ByteMan rule to be used as a temporary measure,
+to prevent the trace information being reported to the APM server being instrument itself. The `Tracer`
+uses Java's `HttpURLConnection` to report the information to the server, and therefore must instruct the
+agent to ignore these communications. This is achieved using the rule defined in the service's
+`src/main/resources/otagent/hawkular-apm.btm` file, with content:
+
+```
+RULE Hawkular APM Ignore server communications
+CLASS java.net.URL
+METHOD openConnection
+HELPER io.opentracing.contrib.agent.OpenTracingHelper
+AT EXIT
+IF $0.getPath().startsWith("/hawkular/apm")
+DO
+  $!.setRequestProperty("opentracing.ignore","true");
+ENDRULE
+```
+
+NOTE: Once the OpenTracing Agent has been released, the Hawkular APM project will be updated to mark the
+`HttpURLConnection` directly with this request property.
+
+### Run the instrumented example
+
+Before being able to try the OpenTracing Java Agent, it will be necessary to obtain the agent jar.
+
+It will then be necessary to [startup a Hawkular APM server](https://hawkular.gitbooks.io/hawkular-apm-user-guide/content/quickstart/) and set up the `HAWKULAR_APM_USER`, `HAWKULAR_APM_PASSWORD` and `HAWKULAR_APM_URI` environment
+variables accordingly within each of the command windows that will run the services.
+
+To run the instrumented version of the services, simply add the agent to the JVM args:
+
+```
+cd servicea
+mvn spring-boot:run -Drun.jvmArguments=-javaagent:<path-to>/opentracing-agent.jar
+```
+
+```
+cd serviceb
+mvn spring-boot:run -Drun.jvmArguments=-javaagent:<path-to>/opentracing-agent.jar
+```
+
+Then re-run the client:
+
+```
+curl http://localhost:8080/greeting -d name=Fred
+```
+
+Finally, start up the Hawkular APM UI and you should see the interactions between the two services on the
+Distributed Tracing page.
```

Comment:
Could you add a screenshot with the expected outcome? I got this and I'm not sure I'm supposed to see more things:

![image](https://cloud.githubusercontent.com/assets/13387/23503458/2ebc4362-ff3c-11e6-8efa-cb48e681d610.png)

![image](https://cloud.githubusercontent.com/assets/13387/23503478/3a26832a-ff3c-11e6-9c48-7f8e57902e4f.png)